### PR TITLE
Organize existing code for editor plugins

### PR DESCRIPTION
**Pull request by aaronfranke**
I was annoyed that the base class for editor plugins wasn't in the `editor/plugins/` folder, nor were the files for changing editor plugin settings or editing project plugins. This PR moves these files into this folder.

Aside from moving files, this PR also modifies `PluginConfigDialog` to have the logic for selecting a language and script name organized together, instead of being mixed with the name/folder/description/author/version logic. This was the main motivation for this PR, I already have another PR lined up which needs this code nicely organized.

~~This PR has one behavior change: In the current master, when creating a new plugin, it will create a new script. However, if you change the file name of a plugin's script, it will not create a new one with the new name. Now it does. Note that it will only create a file if the file is missing, so it will never overwrite existing files.~~

### CODE REVIEW

**Comment by KoBeWi** [comment]
>  if you change the file name of a plugin's script, it will not create a new one with the new name. Now it does.

I think it should first try to rename/move existing file.

File: editor/plugins/plugin_config_dialog.cpp [hunk 1]
```diff
@@ -58,29 +59,34 @@ void PluginConfigDialog::_on_confirmed() {
...
 	cf->save(path.path_join("plugin.cfg"));
+	EditorNode::get_singleton()->get_project_settings()->update_plugins();
+	EditorFileSystem::get_singleton()->scan();
+	_clear_fields();
+}
 
-	if (!_edit_mode) {
+void PluginConfigDialog::_create_script_for_plugin(const String &p_plugin_path, Ref<ConfigFile> p_config_file, int p_script_lang_index) {
+	ScriptLanguage *language = ScriptServer::get_language(p_script_lang_index);
+	ERR_FAIL_COND(language == nullptr);
```
Comment (KoBeWi) [inline editor/plugins/plugin_config_dialog.cpp:80]:
```suggestion
	ERR_FAIL_NULL(language);
```

**Comment by aaronfranke** [comment]
@KoBeWi I don't know the correct way to rename a script in the editor. If I just rename it using DirAccess, the script editor is not informed of the change, so the edited script has the wrong file name still.

**Comment by KoBeWi** [comment]
You can call `update_file()` in EditorFileSystem, or in worst case, just do `rescan()`.

**Comment by aaronfranke** [comment]
@KoBeWi That does not inform the script editor of the change.

**Comment by KoBeWi** [comment]
And if you use this?
https://github.com/KoBeWi/godot/commit/7c165a16cee88c03012ff20e5ba72d7dac9737f5

**Comment by aaronfranke** [comment]
@KoBeWi That still does not work, and it gives a new error too:

Screenshot 2024-04-24 at 9 39 36 PM

**Comment by KoBeWi** [comment]
I was going to look into it, but mixing the behavior change with such enormous refactor makes it unreadable. You should split this into 2 commit.

**Comment by aaronfranke** [comment]
@KoBeWi I updated the PR to remove the behavior change.

File: editor/plugins/plugin_config_dialog.cpp [hunk 2]
```diff
@@ -90,32 +96,21 @@ void PluginConfigDialog::_on_confirmed() {
 		Ref<Script> scr = language->make_template(template_content, class_name, "EditorPlugin");
 		scr->set_path(script_path, true);
 		ResourceSaver::save(scr);
-
-		emit_signal(SNAME("plugin_ready"), scr.ptr(), active_edit->is_pressed() ? _to_absolute_plugin_path(_get_subfolder()) : "");
-	} else {
-		EditorNode::get_singleton()->get_project_settings()->update_plugins();
+		if (!_edit_mode) {
```
Comment (KoBeWi) [inline editor/plugins/plugin_config_dialog.cpp:99]: This check is already done above.

**Review by KoBeWi — APPROVED** [review]
Plugin config dialog changes look fine.

**Review by akien-mga — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

File: editor/plugins/plugin_config_dialog.cpp (hunk 1 above)
Comment (AThousandShips) [reply editor/plugins/plugin_config_dialog.cpp:81]: This was missed

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**